modules/auxiliary/scanner/scada/pcomclient.rb

```diff
@@ -0,0 +1,197 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::Tcp
+  include Rex::Socket::Tcp
+  include Rex::Text
+
+  def initialize(info = {})
+    super(update_info(info,
+                      'Name'          => 'Unitronics PCOM client',
+                      'Description'   => %q{
+        Unitronics Vision PLCs allow authenticated PCOM commands
+        to query PLC registers.
+        This module was inspired in 'modbusclient.rb'},
+          'Author'         => [ 'Luis Rosa <lmrosa[at]dei.uc.pt>' ],
+          'License'        => MSF_LICENSE,
+          'References'     =>
+        [
+          [ 'URL', 'https://unitronicsplc.com/Download/SoftwareUtilities/Unitronics%20PCOM%20Protocol.pdf' ]
+        ],
+        'Actions'        =>
+        [
+          ['READ', { 'Description' => 'Read values from PLC memory' } ],
+          ['WRITE', { 'Description' => 'Write values to PLC memory' } ],
+        ],
+        'DefaultAction' => 'READ'
+                     ))
+    register_options(
+      [
+        Opt::RPORT(20256),
+        OptInt.new('UNITID', [ false, 'Unit ID (0 - 127)', 0]),
+        OptInt.new('ADDRESS', [true, "PCOM memory address (0 - 65535)", 0]),
+        OptInt.new('LENGTH', [true, "Number of values to read (1 - 255) (read only)", 3]),
+        OptString.new('VALUES', [false, "Values to write (0 - 65535 each) (comma separated) (write only)"]),
+        OptEnum.new("OPERAND", [true, 'Operand type', "MI", ["Input", "Output", "SB", "MB", "MI", "SI", "ML", "SL"]]),
```

Comment:
Very small detail, but that extra `,` isn't needed :-)

Reply:
done